File: app/src/androidTest/java/com/fullmeadalchemist/mustwatch/util/TestUtil.java
package com.fullmeadalchemist.mustwatch.util;

import com.fullmeadalchemist.mustwatch.vo.Batch;
import com.fullmeadalchemist.mustwatch.vo.Group;
import com.fullmeadalchemist.mustwatch.vo.User;

import java.util.ArrayList;
import java.util.GregorianCalendar;
import java.util.HashSet;
import java.util.List;
import java.util.Set;



public class TestUtil {

    public static User createUser() {
        String name = "test_user" + randInt(1, 100);
        String email = name + "@example.com";
        return new User(name, email);
    }

    public static List<User> createUsers(int n) {
        Set<Integer> usedIds = new HashSet<>();
        List<User> usersToAdd = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            int userN = randInt(1, 10*n);
            while (usedIds.contains(userN)){
                userN = randInt(1, 10*n);
            }
            usedIds.add(userN);
            String name = "test_user" + userN;
            String email = name + "@example.com";
            usersToAdd.add(new User(name, email));
        }
        return usersToAdd;
    }


    public static Batch createBatch(long userId) {
        Batch b = new Batch();
        b.setName("Dummy Batch #" + randInt(1, 1000));
        b.setCreateDate(GregorianCalendar.getInstance());
        b.setStatus(getStatus());
        b.setTargetSgStarting(roundThreeDecimalPlaces(randDouble(1.10d, 1.30d)));
        b.setTargetSgFinal(roundThreeDecimalPlaces(randDouble(0.95d, 1.05d)));
        b.setStartingPh(roundTwoDecimalPlaces(randFloat(3.0f, 5.5f)));
        b.setStartingTemp(roundOneDecimalPlace(randFloat(65f, 75f)));
        b.setOutputVolume(getRandVolume());
        b.setTargetABV(roundTwoDecimalPlaces(randFloat(0.08f, 0.17f)));
        b.setUserId(userId);
        b.setNotes(String.format("Dummy Batch \"%s\"", b.getName()));
        return b;
    }

    public static Group createGroup(String name) {
        return new Group(name);
    }

//    public static List<Repo> createRepos(int count, String owner, String name,
//                                         String description) {
//        List<Repo> repos = new ArrayList<>();
//        for(int i = 0; i < count; i ++) {
//            repos.add(createRepo(owner + i, name + i, description + i));
//        }
//        return repos;
//    }
//
//    public static Repo createRepo(String owner, String name, String description) {
//        return createRepo(Repo.UNKNOWN_ID, owner, name, description);
//    }
//
//    public static Repo createRepo(int id, String owner, String name, String description) {
//        return new Repo(id, name, owner + "/" + name,
//                description, new Repo.Owner(owner, null), 3);
//    }
//
//    public static Contributor createContributor(Repo repo, String login, int contributions) {
//        Contributor contributor = new Contributor(login, contributions, null);
//        contributor.setRepoName(repo.name);
//        contributor.setRepoOwner(repo.owner.login);
//        return contributor;
//    }
}

File: app/src/androidTest/java/com/fullmeadalchemist/mustwatch/util/TestUtil.kt
package com.fullmeadalchemist.mustwatch.util

import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.randDouble
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.randFloat
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.randInt
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.randVolume
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.roundOneDecimalPlace
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.roundThreeDecimalPlaces
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.roundTwoDecimalPlaces
import com.fullmeadalchemist.mustwatch.demo.DemoGenerators.status
import com.fullmeadalchemist.mustwatch.vo.Batch
import com.fullmeadalchemist.mustwatch.vo.Group
import com.fullmeadalchemist.mustwatch.vo.User
import java.util.*


object TestUtil {

    fun createUser(): User {
        val name = "test_user" + randInt(1, 100)
        val email = "$name@example.com"
        return User(UUID.randomUUID(), name, email)
    }

    fun createUsers(n: Int): List<User> {
        val usedIds = HashSet<Int>()
        val usersToAdd = ArrayList<User>()
        for (i in 0 until n) {
            var userN = randInt(1, 10 * n)
            while (usedIds.contains(userN)) {
                userN = randInt(1, 10 * n)
            }
            usedIds.add(userN)
            val name = "test_user$userN"
            val email = "$name@example.com"
            usersToAdd.add(User(UUID.randomUUID(), name, email))
        }
        return usersToAdd
    }


    fun createBatch(userId: UUID): Batch {
        val b = Batch()
        b.name = "Dummy Batch #" + randInt(1, 1000)
        b.createDate = GregorianCalendar.getInstance()
        b.status = status
        b.targetSgStarting = roundThreeDecimalPlaces(randDouble(1.10, 1.30))
        b.targetSgFinal = roundThreeDecimalPlaces(randDouble(0.95, 1.05))
        b.startingPh = roundTwoDecimalPlaces(randFloat(3.0f, 5.5f))
        b.startingTemp = roundOneDecimalPlace(randFloat(65f, 75f))
        b.outputVolume = randVolume
        b.targetABV = roundTwoDecimalPlaces(randFloat(0.08f, 0.17f))
        b.userId = userId
        b.notes = String.format("Dummy Batch \"%s\"", b.name)
        return b
    }

    fun createGroup(name: String): Group {
        return Group(name)
    }

}